blockchain-agent-proof/src/main/java/com/sphereon/alfresco/blockchain/agent/proof/tasks/ProofVerificationTask.java

```diff
@@ -0,0 +1,56 @@
+package com.sphereon.alfresco.blockchain.agent.proof.tasks;
+
+import com.sphereon.alfresco.blockchain.agent.proof.ProofApiUtils;
+import com.sphereon.alfresco.blockchain.agent.rest.model.VerifyContentAlfrescoResponse;
+import com.sphereon.alfresco.blockchain.agent.tasks.ondemand.VerifyRegistrationTask;
+import com.sphereon.alfresco.blockchain.agent.utils.Signer;
+import com.sphereon.libs.authentication.api.TokenRequest;
+import com.sphereon.sdk.blockchain.proof.api.VerificationApi;
+import com.sphereon.sdk.blockchain.proof.handler.ApiException;
+import com.sphereon.sdk.blockchain.proof.model.ContentRequest;
+import com.sphereon.sdk.blockchain.proof.model.VerifyContentResponse;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+@Component
+@Profile("sphereon-proof-api")
+public class ProofVerificationTask implements VerifyRegistrationTask {
+    private final VerificationApi bcProofVerificationApi;
+    private final TokenRequest tokenRequester;
+    private final Signer signer;
+    private String proofApiConfigName;
+    private ProofApiUtils utils;
+
+    public ProofVerificationTask(final VerificationApi bcProofVerificationApi,
+                                 final TokenRequest tokenRequester,
+                                 final Signer signer,
+                                 final ProofApiUtils utils,
+                                 @Value("${blockchain.config-name:#{null}}") final String proofApiConfigName) {
+        this.bcProofVerificationApi = bcProofVerificationApi;
+        this.tokenRequester = tokenRequester;
+        this.signer = signer;
+        this.utils = utils;
+        this.proofApiConfigName = proofApiConfigName;
+    }
+
+    @Override
+    public VerifyContentAlfrescoResponse verify(final byte[] contentHash) {
+        return this.utils.toAlfrescoResponse(verifyHash(contentHash));
+    }
+
+    private VerifyContentResponse verifyHash(final byte[] contentHash) {
+        final var contentRequest = new ContentRequest();
```

Comment:
I am missing the tokenRequester execution which is present on the register task here.

Not sure how many tasks there are, but maybe use a template method?

Reply:
Hmm, it is a few lines down. I probably moved it here to be closer to the actual API invocation. Still, I don't like this pattern we have here where you have to prepare the authentication and then separately call the API. It should automatically happen if applicable.

I already saw a similar pattern in the Alfresco repository where there is a stateful Alfresco service (see the usage of `AlfrescoRepository::updateCredentials` in combination with an `ObjectFactory`). This looks like a similar pattern. I'll take a closer look at this at a later time.

Reply:
Tokenrequester is configured in `BlockchainProofApiConfig` and used in both the `ProofRegisterTask` and `ProofVerifyTask`. This comment should be resolved with that.